dvc/stage.py

```diff
@@ -774,7 +775,10 @@ def _run(self):
             env=fix_env(os.environ),
             executable=executable,
         )
+
+        signal.signal(signal.SIGINT, signal.SIG_IGN)
```

Comment:
Let's reuse old signal handler that is returned here and wrap all this into try-finally, just in case.
```suggestion
        old_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
        try:
            p.communicate()
        finally:
            signal.signal(signal.SIGINT, old_handler)
```

btw, IIRC, process starts when right when Popen() is called, so there is going to be a brief moment between Popen and here when dvc might receive SIGINT as well, even though it was meant for the child. Maybe we should setup signal handler and then Popen with an old signal handler(using preexec_fn)?

Reply:
E.g.
```
old_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)

def restore_signal_handler():
    signal.signal(signal.SIGINT, old_handler)

try:
    Popen(..., preexeec_fn=restore_signal_handler)
    p.communicate()
finally:
    restore_signal_handler()
```

Reply:
@efiop Thank you for your comments. Nice catch with old signal persistence and unnecessary "race condition". I fixed those and refreshed tests.

I'm a little bit confused about transferring signal policy from dvc process to the child process. If I get it right, a Popen() call doesn't transfer signal policy by default. So, I decided not to use preexeec_fn. Correct me, please, if I'm wrong.